presto-spi/src/main/java/com/facebook/presto/spi/memory/ArrayPool.java

```diff
@@ -16,18 +16,60 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
 
-public class ByteArrayPool
+import static java.util.Objects.requireNonNull;
+
+public class ArrayPool<ArrayType>
 {
+    public static abstract class Allocator<ArrayType>
+    {
+        abstract ArrayType alloc(int size);
+        abstract void init(ArrayType array);
+        abstract int getLength(ArrayType array);
```

Comment:
alloc method takes parameter named `size`, but this method is named `getLength`; perhaps, rename to getSize for consistency